src/BoundsInference.cpp

```diff
@@ -556,12 +561,18 @@ class BoundsInference : public IRMutator {
 
             } else {
                 const vector<Expr> &exprs = consumer.exprs;
-                for (size_t j = 0; j < exprs.size(); j++) {
-                    map<string, Box> new_boxes = boxes_required(exprs[j], scope, func_bounds);
-                    for (const pair<string, Box> &i : new_boxes) {
-                        merge_boxes(boxes[i.first], i.second);
+                Stmt wrapped = Evaluate::make(Call::make(Int(32), "dummy", exprs, Call::PureIntrinsic));
```

Comment:
I think this code should have a comment - the dummy call is surprising. It looks like it's just a way to get an expression that depends on all of the consumer exprs?